sklearn/preprocessing/_polynomial.py

```diff
@@ -346,8 +368,23 @@ def transform(self, X):
         for i in range(n_features):
             spl = self.bsplines_[i]
 
-            if self.extrapolation in ("continue", "error"):
-                XBS[:, (i * n_splines):((i + 1) * n_splines)] = spl(X[:, i])
+            if self.extrapolation in ("continue", "error", "periodic"):
+
+                if self.extrapolation == "periodic":
+                    # With periodic extrapolation we map x to the segment
+                    # [spl.t[k], spl.t[n]].
+                    # This is equivalent to BSpline(.., extrapolate="periodic")
+                    # for scipy>=1.6.1.
+                    n = spl.t.size - spl.k - 1
+                    # Assign to new array to avoid inplace operation
+                    x = spl.t[spl.k] + (X[:, i] - spl.t[spl.k]) % (
+                        spl.t[n] - spl.t[spl.k]
+                    )
```

Comment:
Periodic BSplines were implemented in scipy 1.0.0 in https://github.com/scipy/scipy/pull/7185. If we were to increment our minimal scipy from 0.19 to 1.0 for our next release, scikit-learn 1.0, we would not need this code.

Reply:
The comment was incorrect. 1.6.1 is the current version. I've changed it.